pydarn/exceptions/pydmap_exceptions.py

```diff
@@ -6,16 +6,81 @@
 import logging
 pydarn_logger = logging.getLogger('pydarn')
 
+class CursorError(Exception):
+    """
+    Raise if the cursor is not correctly set
+    """
+    def __init__(self,cursor, expected_value=0, mesage=''):
+        self.cursor
+        if message == '':
+            self.message = "Error: Cursor is at {cursor} and"\
+                    "it needs to be {expected}".format(cursor=cursor,
+                                                       expected=expected_value)
+        else:
+            self.message=message
+        super().__init__(self.message)
+        pydarn_logger.error(self.message)
+
+
 class EmptyFileError(Exception):
     """
     Raised if a file is empty
     """
     def __init__(self,filename):
         self.filename = filename
-        self.message = "Error: {} is empty,"\
+        self.message = "Error: {} is empty or"\
                 " please check this is the correct file".format(filename)
         super().__init__(self.message)
-        pydarn_logger.error("EmptyFileError: {} is empty".format(filename))
+        pydarn_logger.error(self.message)
+
+
+class DmapDataTypeError(Exception):
+    """
+    Raised if the data type is not correct.
+    """
+    def __init__(self, filename, data_name, data_type, cursor):
+        self.filename = filename
+        self.message = "Error: Dmap data type {data_type} for {name}"\
+                " at cursor = {cursor} does not exist in dmap data types."\
+                "filename: {filename}".format(name=data_name,
+                                              data_type=data_type,
+                                              filename=filename,
+                                              cursor=cursor)
+        super().__init__(self.message)
+        pydarn_logger.error(self.message)
+
+
+class ZeroByteError(Exception):
```

Comment:
Personally, I like `Zero` because it is concise and something may have set it to zero which might not have made the bytes invalid. However, I get what you mean with the less than zero cases, so I added `NegativeByteError` which could indicate something really bad happened :/